etw/cs/traceloggingdynamic/TraceLoggingDynamic.cs

```diff
@@ -1579,14 +1711,41 @@ public void AddCountedBinaryArray(string name, Byte[][] values, EventOutType out
         /// fieldCount logical fields should be considered as members of this field.
         /// Note that fieldCount must be in the range 1 to 127.
         /// </summary>
-        public void AddStruct(string name, byte fieldCount, int tag = 0)
+        /// <returns>
+        /// The offset of the struct size within the metadata. This value is normally
+        /// ignored. However, if you do not know the number of fields in a structure
+        /// ahead of time, you can use this value with SetStructFieldCount.
+        /// </returns>
+        public int AddStruct(string name, byte fieldCount, int tag = 0)

```

Comment:
Note: AddStruct now returns the position of the fieldCount value, which can be used in a later call to SetStructFieldCount in case you don't know the struct size until after you've added the fields.